packages/ui/src/components/popover/popover.tsx

```diff
@@ -0,0 +1,198 @@
+
+export enum PopoverVerticalPositions {
+  Top = 'top',
+  Bottom = 'bottom',
+}
+export enum PopoverHorizontalPositions {
+  Left = 'left',
+  Right = 'right',
+}
+export enum PopoverTriggerMode {
+  Click = 'click',
+  Hover = 'hover',
+}
+
+const CLOSE_ON_HOVER_DELAY = 500;
+
+const defaultProps = {
+  verticalPosition: PopoverVerticalPositions.Top,
+  horizontalPosition: PopoverHorizontalPositions.Left,
+  triggerMode: PopoverTriggerMode.Click,
+  className: 'popover',
+  closeOnEsc: true,
+};
+
+interface PopoverProps {
+  onClose?: () => void;
+  onOpen?: () => void;
+  verticalPosition: PopoverVerticalPositions;
+  horizontalPosition: PopoverHorizontalPositions;
+  className: string;
+  closeOnEsc: boolean;
+  triggerMode: PopoverTriggerMode;
+  content: JSX.Element;
+  children: JSX.Element | JSX.Element[];
+}
+
+interface PopoverState {
+  open: boolean;
+}
+
+/**
+ * Popover renders popup with a target.
+ * Properties description:
+ *   onOpen - function that will be executed when popover opens;
+ *   onClose - function that will be executed when popover closes;
+ *   verticalPosition - vertical position of popover relative to target ("top" or "bottom"), default - "top";
+ *   horizontalPosition - horizontal position of popover relative to target ("left" or "right"), default - "left";
+ *   className - popover class, can be use for popover styling, default - 'popover';
+ *   closeOnEsc - handle ESC keyboard pressed event and close popover, default - true;
+ *   triggerMode - popover support 2 ways for opening: click (keyPress) and mouse hover ("click" or "hover"), default - "click";
+ *   content - content of popover. Can be any valid Preact node, ex:
+ *      <select>
+ *          <option>Option 1</option>
+ *          ...
+ *          <option>Option n</option>
+ *      </select>
+ *   children - popover target. Can be any valid Preact node, ex:
+ *      <button>
+ *          <i className="icon" />
+ *      </button>
+ */
+
+export class Popover extends Component<PopoverProps, PopoverState> {
+  private _closeTimeout: any = null;
+  private _controlElement = null;
+  static defaultProps = {
+    ...defaultProps,
+  };
+  state = {
+    open: false,
+  };
+
+  private _clearTimeout = () => {
+    clearTimeout(this._closeTimeout);
+    this._closeTimeout = null;
+  };
+
+  private _handleClickOutside = (e: any) => {
+    if (
+      !!this._controlElement &&
+      !this._controlElement.contains(e.target) &&
+      this.state.open
+    ) {
+      this._closePopover();
+    }
+  };
+
+  private _openPopover = () => {
+    const {onOpen} = this.props;
+    this.setState({open: true}, () => {
+      if (onOpen) {
+        onOpen();
+      }
+    });
+  };
+
+  private _closePopover = () => {
+    const {onClose} = this.props;
+    this._clearTimeout();
+    this.setState({open: false}, () => {
+      if (onClose) {
+        onClose();
+      }
+    });
+  };
+
+  private _handleEscButtonPressed = (e: KeyboardEvent) => {
```

Comment:
consider using the new `onClose` api

Reply:
`onClose` API already exists in Popover or you talking about Popover-content?